components/handlers/module_overview.py

```diff
@@ -111,11 +111,13 @@ def GET(self):
         '''
             Retrieve and render all the info of a module
         '''
-        if SESSION['id'] != web.ACCOUNT_LOGIN_SUCCESSFUL:
+        if not session.validate_session():
             raise web.seeother('/login')
         input_data = web.input()
         module_code = input_data.code
         module_info = model.get_module(module_code)
+        if module_info is None:
```

Comment:
this means that it can't find the module right, so it's taking care of the instance where one user deletes a module and another user tries to view it at the same time?